### PR TITLE
[CBRD-23629] Create utility for Java SP server

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-23629

Users can manage Java SP Server as a service utility same as cub_broker, cub_server etc.

- To support javasp, new routine is added in cubrid utility
```
cubrid javasp <start|stop|restart|status> [database-name]
```
- Because javasp is a service utility, it can be registered in service section in `cubrid.conf`. Since javasp is server (database) dependent, if it is registered in the service section, it works only if it is registered in the server section in the same way as the server.
```
//cubrid.conf
[service]
# The list of processes to be started automatically by 'cubrid service start' command
# Any combinations are available with server, broker, manager and heartbeat.
service=server,broker,manager,javasp

# The list of database servers in all by 'cubrid service start' command.
# This property is effective only when the above 'service' property contains 'server' keyword.
server=demodb,test2
...
```
- In CS Mode, a process for JVM  (cub_javasp) will be created by cubrid utiltiy.
- In SA Mode, a thread for JVM will be created in cub_server as before.

### CODE REVIEW

**Review by byungwook-kim — APPROVED** [review]

File: src/jsp/jsp_cl.c [hunk 1]
```diff
@@ -3116,8 +2842,9 @@ jsp_execute_stored_procedure (const SP_ARGS * args)
   if (error != NO_ERROR || is_prepare_call[call_cnt])
     {
       jsp_send_destroy_request (sock_fd);
-      jsp_close_internal_connection (sock_fd);
-      sock_fds[call_cnt] = INVALID_SOCKET;
+      <<<<<<<HEAD jsp_disconnect_server (sock_fd);
+      == == == = jsp_close_internal_connection (sock_fd);
+      >>>>>>>upstream / develop sock_fds[call_cnt] = INVALID_SOCKET;
```
Comment (byungwook-kim) [inline src/jsp/jsp_cl.c:2847]: not properly merged. check your code again.

File: src/executables/javasp.cpp [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+
+#ident "$Id$"
+
+#include "config.h"
+
+#include <errno.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
```
Comment (byungwook-kim) [inline src/executables/javasp.cpp:35]: required signal.h for the macro SIGxxx, i.e, SIGPIPE

File: src/executables/util_service.c [hunk 3]
```diff
@@ -2279,6 +2344,241 @@ process_manager (int command_type, bool process_window_service)
...
+#endif
+	}
+      else
+	{
+	  const char *args[] = { UTIL_JAVASP_NAME, COMMAND_TYPE_STOP, db_name, NULL };
+	  status = proc_execute (UTIL_JAVASP_NAME, args, true, false, false, NULL);
+	  if (status == NO_ERROR)
+	    {
+	      sleep (1);	/* wait to stop */
+	      status = (is_javasp_running (db_name) != JAVASP_SERVER_STOPPED) ? ER_GENERIC_ERROR : NO_ERROR;
```
Comment (shparkcubrid) [inline src/executables/util_service.c:2486]: How about like this?
```
while (1)
{
     sleep (1);	/* wait to stop */
     status = (is_javasp_running (db_name) != JAVASP_SERVER_STOPPED) ? ER_GENERIC_ERROR : NO_ERROR;
     if (status == JAVASP_SERVER_STOPPED || cnt++ >5) break;
}
```

File: src/executables/util_service.c [hunk 4]
```diff
@@ -2279,6 +2344,241 @@ process_manager (int command_type, bool process_window_service)
...
+	    }
+	}
+    }
+  else
+    {
+      status = ER_GENERIC_ERROR;
+      print_message (stdout, MSGCAT_UTIL_GENERIC_NOT_RUNNING_2S, PRINT_JAVASP_NAME, db_name);
+      util_log_write_errid (MSGCAT_UTIL_GENERIC_NOT_RUNNING_2S, PRINT_JAVASP_NAME, db_name);
+    }
+  print_result (PRINT_JAVASP_NAME, status, STOP);
```
Comment (shparkcubrid) [inline src/executables/util_service.c:2496]: is it needed to check status?
```
if (status != NO_ERROR) { print .... }
```

File: src/executables/util_service.c (hunk 3 above)
Comment (hgryoo) [reply src/executables/util_service.c:2486]: I modified as below. It is the same pattern with process_javasp_start ()
```
  static const int wait_timeout = 5;
  int waited_secs = 0;
	  while (status == NO_ERROR && waited_secs < wait_timeout)
	    {
	      sleep (1);	/* wait to stop */
	      status = (is_javasp_running (db_name) != JAVASP_SERVER_STOPPED) ? ER_GENERIC_ERROR : NO_ERROR;
        waited_secs++;
	    }
```

File: src/executables/util_service.c (hunk 4 above)
Comment (hgryoo) [reply src/executables/util_service.c:2496]: it is not needed.

**Review by shparkcubrid — APPROVED** [review]

**Review by byungwook-kim — APPROVED** [review]

**Review by mhoh3963 — APPROVED** [review]

**Pull request merged**